monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -545,28 +543,38 @@ def get_issues():
         logger.info("Issues generated for reporting")
         return issues_dict
 
-    @staticmethod
-    def is_latest_report_exists():
+    @classmethod
+    def get_latest_event_timestamp(cls) -> float:
+        if not cls._agent_event_repository:
+            raise RuntimeError("Agent event repository does not exist")
+
+        # TODO: Add `get_latest_event` to the IAgentEventRepository
+        agent_events = cls._agent_event_repository.get_events()
+        latest_timestamp = max(agent_events, key=lambda event: event.timestamp).timestamp
+
+        return latest_timestamp
+
+    @classmethod
+    def is_latest_report_exists(cls):
         """
         This function checks if a monkey report was already generated and if it's the latest one.
         :return: True if report is the latest one, False if there isn't a report or its not the
         latest.
         """
-        latest_report_doc = mongo.db.report.find_one({}, {"meta_info.latest_monkey_modifytime": 1})
 
-        if latest_report_doc:
-            report_latest_modifytime = latest_report_doc["meta_info"]["latest_monkey_modifytime"]
-            latest_monkey_modifytime = Monkey.get_latest_modifytime()
-            return report_latest_modifytime == latest_monkey_modifytime
+        if cls._report:
+            report_latest_event_timestamp = cls._report["meta_info"]["latest_event_timestamp"]
+            latest_event_timestamp = cls.get_latest_event_timestamp()
+            return report_latest_event_timestamp == latest_event_timestamp
 
         return False
 
-    @staticmethod
-    def get_report():
-        if not ReportService._agent_repository.get_agents():
+    @classmethod
+    def get_report(cls):
+        if not cls._agent_repository.get_agents():
             return {}
 
-        if not ReportService.is_latest_report_exists():
+        if not cls.is_latest_report_exists():
             return safe_generate_regular_report()
```

Comment:
Here we should probably set the report to what is returned by  `safe_generate_regular_report()`

Reply:
This calls generate_report which does that.

Reply:
If this sets the report, why do we need to `return safe_generate_regular_report()`? Why isn't it
```
update_report()
return cls._report
```
?
Methods usually don't do 2 things. Either generate or set